src/mpyl/projects/versioning.py

```diff
@@ -0,0 +1,151 @@
+"""Versioning and upgrade utilities for mpyl projects.
+### Writing an upgrade script
+
+To write an upgrade script, create a new class that inherits from `Upgrader` and
+implements the `upgrade` method. This class should then be added to the `UPGRADERS`
+list in this module.
+
+"""
+import copy
+import pkgutil
+from abc import ABC
+from pathlib import Path
+from typing import Optional, Generator
+
+from deepdiff import DeepDiff
+from ruamel.yaml.compat import ordereddict
+
+from ..utilities.yaml import yaml_to_string, load_for_roundtrip
+
+VERSION_FIELD = "mpylVersion"
+BASE_RELEASE = "1.0.8"
+
+
+def get_releases() -> list[str]:
+    embedded_releases = pkgutil.get_data(__name__, "releases/releases.txt")
+    if not embedded_releases:
+        raise ValueError("File releases/releases.txt not found in package")
+    releases = embedded_releases.decode("utf-8").strip().splitlines()
+    return list(reversed(releases))
+
+
+def get_latest_release() -> str:
+    return get_releases()[0]
+
+
+class Upgrader(ABC):
+    """Base class for upgrade scripts"""
+
+    target_version: str
+
+    def upgrade(self, previous_dict: ordereddict) -> ordereddict:
+        return previous_dict
+
+
+class UpgraderOne11(Upgrader):
+    target_version = "1.0.11"
+
+    def upgrade(self, previous_dict: ordereddict) -> ordereddict:
+        return previous_dict
```

Comment:
The method is implemented in the base class though right? Unless you want to override it it shouldn't be necessary to add it right?

Reply:
https://github.com/Vandebron/mpyl/pull/207#discussion_r1314662223

Reply:
Yes I saw, that's why I commented

Reply:
[[TECH-556] Remove unnecessary override](https://github.com/Vandebron/mpyl/pull/207/commits/2c17b5dfc435c356f16fddb426cb3a8b422a4296)